votor/src/certificate_pool_service.rs

```diff
@@ -206,10 +213,17 @@ impl CertificatePoolService {
 
         // Ingest votes into certificate pool and notify voting loop of new events
         while !ctx.exit.load(Ordering::Relaxed) {
-            // TODO: we need set identity here as well
+            // Update the current pubkey if it has changed
+            let new_pubkey = ctx.cluster_info.id();
+            if my_current_pubkey != new_pubkey {
```

Comment:
A log here would be good, otherwise LGTM

Reply:
Done.